phoenicis-multithreading/src/main/java/org/phoenicis/multithreading/debug/ControlledThreadPoolExecutorDebuggerWatcher.java

```diff
@@ -0,0 +1,81 @@
+package org.phoenicis.multithreading.debug;
+
+import org.apache.commons.lang.mutable.MutableBoolean;
+import org.phoenicis.multithreading.ControlledThreadPoolExecutorService;
+import org.slf4j.Logger;
+
+import java.util.HashMap;
+import java.util.Map;
+import java.util.concurrent.ExecutorService;
+
+/**
+ * Watcher for {@link ControlledThreadPoolExecutorDebugger}
+ *
+ * @see ControlledThreadPoolExecutorDebugger
+ */
+class ControlledThreadPoolExecutorDebuggerWatcher implements Runnable {
+    private final long startTime = System.currentTimeMillis();
+    private final MutableBoolean running;
+    private final ExecutorService[] poolsToObserve;
+    private final Logger logger;
+    private final int sleepTime;
+    private final Map<ExecutorService, Long> lastNumberOfTasks = new HashMap<>();
+    private final Map<ExecutorService, Long> lastTime = new HashMap<>();
+
+    ControlledThreadPoolExecutorDebuggerWatcher(MutableBoolean running,
+            ExecutorService[] poolsToObserve,
+            Logger logger,
+            int sleepTime) {
+        this.running = running;
+        this.poolsToObserve = poolsToObserve;
+        this.logger = logger;
```

Comment:
Why is it required to pass the logger?

Reply:
So that we can write a unit test for this component

Reply:
I see. As it's debugging code, I think the approach is valid.